runtime/common/src/crowdloan.rs

```diff
@@ -304,23 +313,23 @@ decl_module! {
 		/// slot. It will be withdrawable in two instances: the parachain becomes retired; or the
 		/// slot is unable to be purchased and the timeout expires.
 		#[weight = 0]
-		fn contribute(origin, #[compact] index: FundIndex, #[compact] value: BalanceOf<T>) {
+		fn contribute(origin, #[compact] index: FundIndex, #[compact] value: BalanceOf<T>, return_to: T::AccountId) {
```

Comment:
i don't like this change. it facilitates a "shadow transfer", i.e. allowing transfer of funds between accounts while being ostensibly for some other use. this means you can't e.g. introduce the crowdfund module when a chain (for legal reasons) cannot enable transfers, and you can't have a proxy type which allows crowdloans to be made but disallowed transfers.

Reply:
Just requires transfer permission in order for proxy to participate crowdloan. Otherwise because proxy setup will not exist on parachain, it will simply make the rewards inaccessible.